src/messaging/ExchangeMgr.h

```diff
@@ -0,0 +1,266 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      Defines the CHIP ExchangeManager class and its supporting types
+ *      for Exchange management.
+ *
+ */
+
+#pragma once
+
+#include <messaging/ExchangeContext.h>
+#include <support/DLLUtil.h>
+#include <transport/SecureSessionMgr.h>
+
+#define EXCHANGE_CONTEXT_ID(x) ((x) + 1)
+
+using namespace chip::Transport;
+
+namespace chip {
+
+using System::PacketBuffer;
+
+class ExchangeContext;
+
+static constexpr int16_t ANY_MESSAGE_TYPE = -1;
```

Comment:
nit: I would consider kAnyMessageType rather than all case (generally ALL_CASE seem to be macros).
Then again, I am not sure we have any convention here, so it should not matter.

Reply:
Acked.